ocaml/xenopsd/xc/device_common.ml

```diff
@@ -553,6 +553,7 @@ let qmp_send_cmd ?send_fd domid cmd =
           result
       | message ->
           let msg' = Qmp.string_of_message message in
-          error "QMP result for domid %d: %s (%s)" domid msg' __LOC__ ;
+          if may_fail = Some true then
```

Comment:
wouldn't you want to skip the error message when you expect a failure, does this need to be `if not may_fail then error...`

Reply:
It's when the caller expects the call to fail that it doesn't make sense to log the failure there, this is because on the caller side the failure is accounted for and there is no need to have logs to debug the situation. It's only in cases where the failure is not expected that it's beneficial to log it.